### ISSUE TITLE
Supporting multiple signature_def's at runtime

### ISSUE THREAD

**Issue by raimondasl** [triton-inference-server/server#2331]
TensorFlow 2 tf.saved_model.save() saves each @tf.function decorated function into a separate signature_def. So for models with multiple operations/functions - for example, for stateful inference or for training - TensorFlow saved model contains multiple signature_defs which is not supported by Triton. For example, this is what I get for model with serve/train/get_weights/set_weights operations:
Code:

```
    tf.saved_model.save(module, model_dir,
        signatures={
            'serve' :
            module.serve.get_concrete_function({'input_ids':tf.TensorSpec([None, None], tf.int32),
                                                'external_features':tf.TensorSpec([None, None], tf.float32),
                                               }),
            'train' :
            module.train.get_concrete_function({'input_ids':tf.TensorSpec([None, None], tf.int32), \
                                   'external_features':tf.TensorSpec([None, None], tf.float32),
                                  },
                                  {'intent_ids':tf.TensorSpec([None], tf.int32)}),

            'get_weights' : module.get_weights.get_concrete_function(tf.TensorSpec([None], tf.float32)),
            'set_weights' : module.set_weights.get_concrete_function(tf.TensorSpec([None], tf.float32)),
        })
```

Result:
```
$ saved_model_cli show --dir saved_model_dir/ --tag_set serve
The given SavedModel MetaGraphDef contains SignatureDefs with the following keys:
SignatureDef key: "__saved_model_init_op"
SignatureDef key: "get_weights"
SignatureDef key: "serve"
SignatureDef key: "set_weights"
SignatureDef key: "train"
```

It is possible to put all the functionality (serve/train/get_weights/set_weights) into a single @tf.function that takes all inputs and produces all outputs and select what is actually executed based on selector input. But this is hacky and ugly.
In TensorFlow 1, we could save all the outputs to be accessed from Triton into single signature_def via tf.compat.v1.saved_model.simple_save(). But though tf.compat.v1.saved_model.simple_save() is still "supported" in TF2, it does not work with TF2 variables.

In summary, with TensorFlow 2 it would be very beneficial for Triton to support multiple signature_defs during runtime. The infer request sent to Triton would have to indicate the signature_def to be used.

The example comes from a toy BERT-like model with the signatures I posted. If needed, I could share the Python Jupiter notebook or Python script with this toy model.

As discussed directly with Triton team, this is somewhat high priority issue.

**Comment by dzier** [comment]
Fixed in https://github.com/triton-inference-server/tensorflow_backend/pull/21 and https://github.com/triton-inference-server/tensorflow_backend/pull/23
